File: backend-api/src/main/java/demo/backend_api/dto/StockResponse.java
package demo.backend_api.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class StockResponse {
    private String symbol;
    private String companyName;
    private String sector;
    private Double currentPrice;
    private Double previousClose;
    private Double change;
    private Double changePercent;
}

File: frontend/src/app/models/stock-response.ts
export interface StockResponse {
  symbol: string;
  companyName: string;
  sector: string;
  currentPrice: number;
  previousClose: number;
  change: number;
  changePercent: number;
  isActive?: boolean;
}

export interface PriceHistoryResponse {
  recordedDate: string; 
  closePrice: number;
  openPrice?: number;
  highPrice?: number;
  lowPrice?: number;
  historyId?: number;
  stock?: StockResponse;
}

export type TimeRange = '1d' | '1w' | '1m' | '6m' | '1y';

export interface TodaysPrice {
  id?: number;
  symbol: string;
  price: number;
  volume: number;
  timestamp: string;
}